### PR TITLE
Android agent v7.0.0 release notes & doc updates

### PR DESCRIPTION
**Pull request by ywang-nr**
* What problems does this PR solve? - Android agent v7.0.0 release notes & doc updates

1. AGP 8 support
2. Config cache support
3. Compatibility doc update
4. Release note callout to indicate minimum Gradle and AGP version upgrades

### CODE REVIEW

**Comment by ywang-nr** [comment]
@cthomas-newrelic for review

File: src/content/docs/mobile-monitoring/new-relic-mobile-android/get-started/new-relic-android-compatibility-requirements.mdx [hunk 1]
```diff
@@ -55,6 +55,12 @@ HttpURLConnection, OkHttp2 (except versions 2.0 and 2.4), OkHttp (versions 2.8,
   AndroidHttpClient was removed from the SDK in version 23, and as such only supported in Android agent version 6.5.0 and lower.
   </Callout>
 
+**Android agent version 7.0.0 or higher** requires to meet the following requirements:
+
+* Android Studio
+* Gradle 7.1.0
```
Comment (cthomas-newrelic) [inline src/content/docs/mobile-monitoring/new-relic-mobile-android/get-started/new-relic-android-compatibility-requirements.mdx:61]: Gradle `7.0`
Android Gradle Plugin `7.0.0`

File: src/content/docs/release-notes/mobile-release-notes/android-release-notes/android-700.mdx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+subject: Android agent
+releaseDate: '2023-06-08'
+version: 7.0.0
+downloadLink: 'https://download.newrelic.com/android_agent/ant/NewRelic_Android_Agent_7.0.0.zip'
+---
+
+
+## New
+
+* AGP 8 support
```
Comment (cthomas-newrelic) [inline src/content/docs/release-notes/mobile-release-notes/android-release-notes/android-700.mdx:11]: Adds support for Gradle 8.0 and higher
Adds support for the Android Gradle Plugin 8.0 and higher
Adds support for Gradle configuration caching

File: src/content/docs/release-notes/mobile-release-notes/android-release-notes/android-700.mdx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+---
+subject: Android agent
+releaseDate: '2023-06-08'
+version: 7.0.0
+downloadLink: 'https://download.newrelic.com/android_agent/ant/NewRelic_Android_Agent_7.0.0.zip'
+---
+
+
+## New
+
```
Comment (cthomas-newrelic) [inline src/content/docs/release-notes/mobile-release-notes/android-release-notes/android-700.mdx:10]: Add block to announce and describe major version increase: include new support levels, deprecations, caveats, what is no longer supported. Check version 6.0.0 for example

File: src/content/docs/mobile-monitoring/new-relic-mobile-android/get-started/new-relic-android-compatibility-requirements.mdx [hunk 4]
```diff
@@ -55,6 +55,12 @@ HttpURLConnection, OkHttp2 (except versions 2.0 and 2.4), OkHttp (versions 2.8,
   AndroidHttpClient was removed from the SDK in version 23, and as such only supported in Android agent version 6.5.0 and lower.
   </Callout>
 
+**Android agent version 7.0.0 or higher** requires to meet the following requirements:
```
Comment (cthomas-newrelic) [inline src/content/docs/mobile-monitoring/new-relic-mobile-android/get-started/new-relic-android-compatibility-requirements.mdx:58]: Rephrase

File: src/content/docs/mobile-monitoring/new-relic-mobile-android/get-started/new-relic-android-compatibility-requirements.mdx (hunk 1 above)
Comment (cthomas-newrelic) [reply src/content/docs/mobile-monitoring/new-relic-mobile-android/get-started/new-relic-android-compatibility-requirements.mdx:61]: * DexGuard 9.0 and higher

**Review by cthomas-newrelic — CHANGES_REQUESTED** [review]

File: src/content/docs/release-notes/mobile-release-notes/android-release-notes/android-700.mdx (hunk 2 above)
Comment (cthomas-newrelic) [reply src/content/docs/release-notes/mobile-release-notes/android-release-notes/android-700.mdx:11]: Adds support for JDK 17 when used with Gradle 8

File: src/content/docs/mobile-monitoring/new-relic-mobile-android/get-started/new-relic-android-compatibility-requirements.mdx (hunk 4 above)
Comment (ywang-nr) [reply src/content/docs/mobile-monitoring/new-relic-mobile-android/get-started/new-relic-android-compatibility-requirements.mdx:58]: How do we want to rephrase it?

File: src/content/docs/mobile-monitoring/new-relic-mobile-android/get-started/new-relic-android-compatibility-requirements.mdx [hunk 5]
```diff
@@ -55,6 +55,13 @@ HttpURLConnection, OkHttp2 (except versions 2.0 and 2.4), OkHttp (versions 2.8,
...
+
+* Android Studio
+* Gradle 7.0
```
Comment (ywang-nr) [inline src/content/docs/mobile-monitoring/new-relic-mobile-android/get-started/new-relic-android-compatibility-requirements.mdx:61]: Gradle 7.1

File: src/content/docs/mobile-monitoring/new-relic-mobile-android/get-started/new-relic-android-compatibility-requirements.mdx [hunk 6]
```diff
@@ -55,6 +55,13 @@ HttpURLConnection, OkHttp2 (except versions 2.0 and 2.4), OkHttp (versions 2.8,
...
+* Gradle plugin 7.0.0
```
Comment (ywang-nr) [inline src/content/docs/mobile-monitoring/new-relic-mobile-android/get-started/new-relic-android-compatibility-requirements.mdx:62]: Android Gradle plugin

File: src/content/docs/mobile-monitoring/new-relic-mobile-android/get-started/new-relic-android-compatibility-requirements.mdx [hunk 7]
```diff
@@ -55,6 +55,13 @@ HttpURLConnection, OkHttp2 (except versions 2.0 and 2.4), OkHttp (versions 2.8,
   AndroidHttpClient was removed from the SDK in version 23, and as such only supported in Android agent version 6.5.0 and lower.
   </Callout>
 
+**Android agent version 7.0.0 or higher** requires to meet the following requirements:
+
+* Android Studio
```
Comment (ywang-nr) [inline src/content/docs/mobile-monitoring/new-relic-mobile-android/get-started/new-relic-android-compatibility-requirements.mdx:60]: Android studio flamingo + version number

File: src/content/docs/release-notes/mobile-release-notes/android-release-notes/android-700.mdx (hunk 3 above)
Comment (ywang-nr) [reply src/content/docs/release-notes/mobile-release-notes/android-release-notes/android-700.mdx:17]: Consult with doc team

File: src/content/docs/mobile-monitoring/new-relic-mobile-android/install-configure/configure-proguard-or-dexguard-android-apps.mdx [hunk 8]
```diff
@@ -29,6 +29,16 @@ To add support for [ProGuard](http://proguard.sourceforge.net/manual/usage.html)
    -keep class com.newrelic.** { *; }
    -dontwarn com.newrelic.**
    -keepattributes Exceptions, Signature, InnerClasses, LineNumberTable, SourceFile, EnclosingMethod
+
+  ##
+  ## NewRelic Gradle plugin 7.x requires the following additions:
```
Comment (cthomas-newrelic) [inline src/content/docs/mobile-monitoring/new-relic-mobile-android/install-configure/configure-proguard-or-dexguard-android-apps.mdx:34]: Please change "requires" to "may require". Not all apps will encounter this issue.

File: src/content/docs/mobile-monitoring/new-relic-mobile-android/install-configure/configure-proguard-or-dexguard-android-apps.mdx [hunk 9]
```diff
@@ -29,6 +29,16 @@ To add support for [ProGuard](http://proguard.sourceforge.net/manual/usage.html)
...
+  ##
+  # Retain generic signatures of TypeToken and its subclasses with R8 version 3.0 and higher.
```
Comment (cthomas-newrelic) [inline src/content/docs/mobile-monitoring/new-relic-mobile-android/install-configure/configure-proguard-or-dexguard-android-apps.mdx:36]: Please create link of "R8 Version 3.0" with "https://r8.googlesource.com/r8/+/refs/heads/master/compatibility-faq.md#r8-full-mode"

Comment (cthomas-newrelic) [reply src/content/docs/mobile-monitoring/new-relic-mobile-android/install-configure/configure-proguard-or-dexguard-android-apps.mdx:36]: Scratch that: line should read "if R8 version 3.0 full-mode is enabled", using the link provided above.

File: src/content/docs/mobile-monitoring/new-relic-mobile-android/get-started/new-relic-android-compatibility-requirements.mdx [hunk 10]
```diff
@@ -51,15 +51,25 @@ Make sure your Android app meets these requirements:
...
 HttpURLConnection, OkHttp2 (except versions 2.0 and 2.4), OkHttp (versions 2.8, 3.5+, 4.0+), OkIO (version 1.11), AndroidHttpClient, Volley (version 1.0.0), and Apache HTTP Client networking APIs
 
-<Callout variant="important">
+  <Callout variant="important">
   AndroidHttpClient was removed from the SDK in version 23, and as such only supported in Android agent version 6.5.0 and lower.
   </Callout>
 
+**Android agent version 7.0.0 or higher** requires:
+
+* Android Studio Flamingo 2022.2.1 RC 1
```
Comment (cthomas-newrelic) [inline src/content/docs/mobile-monitoring/new-relic-mobile-android/get-started/new-relic-android-compatibility-requirements.mdx:60]: Remove version number`2022.2.1 RC 1`

File: src/content/docs/release-notes/mobile-release-notes/android-release-notes/android-700.mdx [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+version: 7.0.0
+downloadLink: 'https://download.newrelic.com/android_agent/ant/NewRelic_Android_Agent_7.0.0.zip'
+---
+
+<Callout variant="important">
+  Welcome to the New Relic Android agent version 7.0. This major release breaks compatibility with versions of 
+  * Gradle 6.x
+  * Android Gradle Plugin (AGP) 4.x
+
+  Android agent 7.x is increasing the minimum Gradle version to 7.0. Customers will be required to upgrade to Android Gradle Plugin (AGP) 7.0.0 and higher.
```
Comment (cthomas-newrelic) [inline src/content/docs/release-notes/mobile-release-notes/android-release-notes/android-700.mdx:13]: Gradle `7.1`, AGP 7.0.0

File: src/content/docs/release-notes/mobile-release-notes/android-release-notes/android-700.mdx [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+
+## New
+
+* Adds support for Gradle 8.0 and higher
+* Adds support for the Android Gradle Plugin 8.0 and higher
+* Adds support for Gradle configuration caching
+* Adds support for JDK 17 when used with Gradle 8
+
+
+## Fixed
```
Comment (cthomas-newrelic) [inline src/content/docs/release-notes/mobile-release-notes/android-release-notes/android-700.mdx:24]: This is a new major version, so we can remove the `Fixed` section

**Review by cthomas-newrelic — CHANGES_REQUESTED** [review]
A few minor edits, please.

File: src/content/docs/mobile-monitoring/new-relic-mobile-android/get-started/new-relic-android-compatibility-requirements.mdx [hunk 13]
```diff
@@ -51,15 +51,25 @@ Make sure your Android app meets these requirements:
...
   </Callout>
 
+**Android agent version 7.0.0 or higher** requires:
+
+* Android Studio Flamingo
+* Gradle 7.1
+* Android Gradle plugin 7.0.0
+* DexGuard 9.0 and higher
+  <Callout variant="important">
+  DexGuard 9 doesn't yet support Gradle 8 or AGP 8
```
Comment (cthomas-newrelic) [inline src/content/docs/mobile-monitoring/new-relic-mobile-android/get-started/new-relic-android-compatibility-requirements.mdx:65]: You know, let's just pull the call-out. I haven't rechecked the status of DG9 and Gradle 8, and we'd have to remove this anyway when it is supported.

**Review by cthomas-newrelic — CHANGES_REQUESTED** [review]
One more please. Let's pull the DexGuard 9 call-out

File: src/content/docs/mobile-monitoring/new-relic-mobile-android/get-started/new-relic-android-compatibility-requirements.mdx (hunk 13 above)
Comment (ywang-nr) [reply src/content/docs/mobile-monitoring/new-relic-mobile-android/get-started/new-relic-android-compatibility-requirements.mdx]: Resolved

**Review by cthomas-newrelic — APPROVED** [review]
🐑 it

**Comment by ywang-nr** [comment]
Ready for the doc team to merge. Thanks.

**Review by homelessbirds — APPROVED** [review]

**Pull request merged**